README.md

```diff
@@ -1019,6 +1019,255 @@ Voilà! Since we updated `index.html`, you'll need to restart the dev server. Go
+        let status = Fetch.Response.status(response);
+        if (status === 200) {
+          /* If status is OK, proceed to parse the response */
+          response
+          |> Fetch.Response.text
+          |> P.then_(text =>
+              {
+                let data =
+                  try(Ok(text |> Json.parseOrRaise |> Feed.feed_of_json)) {
+                  | Json.Decode.DecodeError(msg) =>
+                    Js.Console.error(msg);
+                    Error("Failed to decode: " ++ msg);
+                  };
+
+                switch (data) {
+                | Error(_) =>
+                  setState(state => {...state, step: Loaded(data)})
+                | Ok(data) =>
+                  setState(state => {
+                    let updatedFeed =
+                      switch (state.step) {
+                      | Loaded(Ok(feed))
+                      | Loading(Some(Ok(feed))) =>
+                        mergeFeeds(feed, data)
+                      | _ => data
+                      };
+                    {
+                      ...state,
+                      step: Loaded(Ok(updatedFeed)),
+                      currentPage: page + 1,
+                    };
+                  })
+                };
+              }
+              |> P.resolve
+            );
+        } else {
+          /* Handle non-200 status */
+          ...
+        };
+      })
+  |> ignore;
+};
+```
+
+#### 11.5: Modify render code to include scroll sentinel and handling new state
+
+We’ll add a `sentinelRef` to track the last element of the list, which will
+trigger loading more data when it comes into view.
+
+In the render code, create a new React ref to track the last element of the
+list:
+
+```reason
+    let sentinelRef = React.useRef(Js.Nullable.null);
+```
+
+Also, add the element itself at the end of the `switch`:
+
+```reason
+  {switch (state.step) {
+    | InvalidUsername => <div> {React.string("Invalid username")} </div>
+    | Idle =>
+      <div>
+        {React.string(
+          "Press the \"Enter\" key to confirm the username selection.",
+        )}
+      </div>
+    | Loading(None | Some(Error(_))) =>
+      <div> {React.string("Loading...")} </div>
+    | Loading(Some(Ok(feed))) =>
+      <> {renderFeed(feed)} <div> {React.string("Loading...")} </div> </>
+    | Loaded(Error(msg)) => <div> {React.string(msg)} </div>
+    | Loaded(Ok(feed)) => renderFeed(feed)
+    }}
+  <div ref={ReactDOM.Ref.domRef(sentinelRef)} />
+```
+
+#### 11.6: Update the effect to fetch new data when observer is intersecting
+
+Finally, let's update the effect to fetch more data when the
+`IntersectionObserver` detects that the sentinel element is in the viewport.
+
+```reason
+  React.useEffect1(
+    () => {
+      switch (Username.make(state.username)) {
+      | Error () =>
+        Js.Exn.raiseError("The value of state.username is invalid")
+      | Ok(username) =>
+        switch (Js.Nullable.toOption(sentinelRef.current)) {
```

Comment:
Might want to explain why we convert to option first instead of directly pattern matching on the Js.Nullable.t  value

Reply:
How would you explain this in an easy way? I'd like to avoid getting into the runtime representation shenanigans for the purpose of the workshop.

Reply:
I think we don't need to give any details, just mention that `Js.Nullable.t` is an opaque type and we can't pattern match on it like we can with `Option.t`

Reply:
https://github.com/ahrefs/react-alicante-workshop/pull/9/commits/e8b32bd42e3de0b8ce83234d86a1ddcaceb519bd